lib/doubly_linked_list.rb

```diff
@@ -0,0 +1,145 @@
+# Defines a node in the doubly linked list
+class DoubleNode
+  attr_reader :data # allow external entities to read value but not write
+  attr_accessor :previous, :next # allow external entities to read or write previous or next node
+
+  def initialize(value, previous_node = nil, next_node = nil)
+    @data = value
+    @previous = previous_node
+    @next = next_node
+  end
+end
+
+# Defines the doubly linked list
+class DoublyLinkedList
+  def initialize
```

Comment:
Nice that you have a doubly linked list, but I would suggest also having a `@tail` attribute.  That would make adding and retrieving elements from the rear of the list much more efficient.